porespy/metrics/__funcs__.py

```diff
@@ -735,23 +738,25 @@ def region_surface_areas(regions, voxel_size=1, strel=None):
         that the surface area of region 1 is stored in element 0 of the list.
 
     """
-    print('-'*60)
-    print('Finding surface area of each region')
+    print('-' * 60, flush=True)
+    print('Finding surface area of each region', flush=True)
     im = regions.copy()
     # Get 'slices' into im for each pore region
     slices = spim.find_objects(im)
     # Initialize arrays
     Ps = np.arange(1, np.amax(im)+1)
     sa = np.zeros_like(Ps, dtype=float)
     # Start extracting marching cube area from im
-    for i in tqdm(Ps):
+    tqdm_Ps = tqdm(Ps, file=sys.stdout)
```

Comment:
If we're going to change the tqdm behavior, it should be:
with tqdm(iterator) as pbar:
   pbar.update()
   <code here>
This way the pbar always closes properly even if the code errors.  Here is an example: https://github.com/PMEAL/porespy/blob/e82c22e3923e6f0b89db85588f5745cc3c1f4f6d/porespy/filters/__funcs__.py#L1305

Reply:
Maybe you can find-all and update the tqdm usage everywhere while you're at it?

Reply:
Yep, I did. I just realized that the issue with tqdm is "not flushing the sys.stdout" rather than tqdm object not getting closed. I tested it and adding `file=sys.stdout` fixes the flush problem. I changed it everywhere.